src/shared/lib/URLWrapper.ts

```diff
@@ -39,6 +52,23 @@ export default class URLWrapper<QueryParamsType> {
         this.routing.updateRoute(query as any);
     }
 
+    private setProperty(property:Property<QueryParamsType>, query:QueryParamsType){
+        this._setPropertyAndHandleUndefined(property, query[property.name]);
+        // if it's still undefined, then check aliases
+        if (this.query[property.name] === undefined && property.aliases && property.aliases.length) {
+            for (const alias of property.aliases) {
+                this._setPropertyAndHandleUndefined(property, query[alias]);
+                // once you've set it, don't bother with any other aliases
+                if (this.query[property.name]) break;
+            }
+        }
+    }
+
+    private _setPropertyAndHandleUndefined(property:Property<QueryParamsType>, value:string|undefined){
+        // @ts-ignore
+        this.query[property.name] = typeof value === "string" ? decodeURIComponent(value) : undefined;
```

Comment:
something just occurred to me - the routing library doesnt URI decoding?

Reply:
i don't think so.  i believe i tested it.

Reply:
and we were doing it in old model (decoding it)